src/ir/ty.rs

```diff
@@ -104,9 +104,13 @@ impl Type {
         Self::new(Some(name), None, kind, false)
     }
 
-    pub fn is_integer_literal(&self) -> bool {
+    pub fn is_integer(&self, ctx: &BindgenContext) -> bool {
         match *self.kind() {
             TypeKind::Int(..) => true,
+            TypeKind::ResolvedTypeRef(inner) |
+            TypeKind::Alias(_, inner) => {
```

Comment:
Yeah, using `canonical_type` should be better.

The issue for adding a test is that... the original test fails even on master for me... :(